### PR TITLE
refactor: squad directory on tablet and mobile

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Should be ready for initial review but I still have some concerns I will relay to the design team. Regarding width size - so on that side, feel free to ignore it.
- Deployed on preview. Quick link: https://preview.app.daily.dev/squads

Previews:
Screenshot 2024-03-20 at 11 51 18 PM
Screenshot 2024-03-20 at 11 51 15 PM
Screenshot 2024-03-20 at 11 51 05 PM
Screenshot 2024-03-20 at 11 51 00 PM
Screenshot 2024-03-20 at 11 50 53 PM

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-132 #done

### CODE REVIEW

File: packages/shared/src/lib/query.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import { SharedFeedPage } from '../components/utilities';
 
 export enum OtherFeedPage {
   Tag = 'tag',
-  Squad = 'squad',
```
Comment (sshanzel) [inline packages/shared/src/lib/query.ts:18]: Was non-existent. After checking, the value in the `feedName` context is `squads`.

File: packages/shared/src/components/squads/SquadsDirectoryHeader.tsx [hunk 2]
```diff
@@ -1,52 +1,59 @@
...
+  className?: string;
+}
+
+export const SquadsDirectoryHeader = ({
+  className,
+}: SquadsDirectoryHeaderProps): ReactElement => {
   const { openNewSquad } = useSquadNavigation();
 
   return (
-    <div className="mb-4">
```
Comment (sshanzel) [inline packages/shared/src/components/squads/SquadsDirectoryHeader.tsx:12]: Diff looks crazy, but the major difference is just the removal of this unnecessary wrapper and allowing to pass `className`.

**Comment by nensidosari** [comment]
Screenshot 2024-03-21 at 9 12 22 AM
Maybe something u're already checking with design, but this was the only thing that caught my eye that doesn't look 100% right. 

The border surrounding the cards, and then no space between the cards itself, I know this is how mobile feed looks like, but here doesn't look that goood on my opinion.

File: packages/webapp/pages/squads/index.tsx [hunk 3]
```diff
@@ -15,15 +15,33 @@ import {
   SourceCard,
```
Comment (nensidosari) [inline packages/webapp/pages/squads/index.tsx:15]: Feel like our components always end up with lots of lines of code making it harder to read....

**Review by nensidosari — APPROVED** [review]
Looks good to me

**Review by rebelchris — CHANGES_REQUESTED** [review]
Agree with Nensi, spacing on mobile for public squads looks very odd.
I'll block it until we resolve that 🙏

File: packages/webapp/pages/squads/index.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/webapp/pages/squads/index.tsx:15]: We can probably break this file down in a separate PR. Would be glad to do that.

**Comment by sshanzel** [comment]
After checking with the design team, it seems we should not use full-width on single columns.
[Thread](https://dailydotdev.slack.com/archives/C0650M2KX8E/p1711012731064069?thread_ts=1710996225.906289&cid=C0650M2KX8E)

Screenshot 2024-03-21 at 5 22 18 PM

**Comment by sshanzel** [comment]
Screenshot 2024-03-21 at 5 24 34 PM

**Review by rebelchris — APPROVED** [review]
Now it looks fine indeed!

**Review by nensidosari — APPROVED** [review]
Looks good to me now as well

**Pull request merged**